EIPS/eip-4626.md

```diff
@@ -379,10 +379,12 @@ MUST emit the `Withdraw` event.
 
 MUST support a withdraw flow where the shares are burned from `owner` directly where `owner` is `msg.sender`.
 
-MUST support a withdraw flow where the shares are burned from `owner` directly where `msg.sender` has ERC-20 approval over the shares of `owner`.
+MUST support a withdraw flow where the shares are burned from `owner` directly where `msg.sender` has EIP-20 approval over the shares of `owner`.
 
 MAY support an additional flow in which the shares are transferred to the Vault contract before the `withdraw` execution, and are accounted for during `withdraw`.
 
+MUST check msg.sender can spend owner funds, assets needs to be converted to shares and shares should be checked for allowance.
```

Comment:
```suggestion
SHOULD check `msg.sender` can spend owner funds, assets needs to be converted to shares and shares should be checked for allowance.
```
I would be okay approving this if this uses should, as vulnerable contracts would still be compliant.